jffs2/libjffs2.c

```diff
@@ -918,236 +975,86 @@ static int jffs2_srv_statfs(jffs2_partition_t *p, void *buf, size_t len)
 }
 
 
-int jffs2lib_message_handler(void *partition, msg_t *msg)
+static int libjffs2_devctl(void *info, oid_t *oid, const void *in, void *out)
 {
-	jffs2_partition_t *p = partition;
-
-	switch (msg->type) {
-	case mtOpen:
-		msg->o.io.err = jffs2_srv_open(p, &msg->i.openclose.oid);
-		break;
-
-	case mtClose:
-		msg->o.io.err = jffs2_srv_close(p, &msg->i.openclose.oid);
-		break;
-
-	case mtRead:
-		msg->o.io.err = jffs2_srv_read(p, &msg->i.io.oid, msg->i.io.offs, msg->o.data, msg->o.size);
-		break;
-
-	case mtWrite:
-		msg->o.io.err = jffs2_srv_write(p, &msg->i.io.oid, msg->i.io.offs, msg->i.data, msg->i.size);
-		break;
-
-	case mtTruncate:
-		msg->o.io.err = jffs2_srv_truncate(p, &msg->i.io.oid, msg->i.io.len);
-		break;
-
-	case mtDevCtl:
-		msg->o.io.err = -EINVAL;
-		break;
-
-	case mtCreate:
-		msg->o.create.err = jffs2_srv_create(p, &msg->i.create.dir, msg->i.data, msg->i.size, &msg->o.create.oid, msg->i.create.type, msg->i.create.mode, &msg->i.create.dev);
-		break;
-
-	case mtDestroy:
-		msg->o.io.err = jffs2_srv_destroy(&msg->i.destroy.oid);
-		break;
+	return -EINVAL;
```

Comment:
I missed this one, maybe ENOSYS here too?

Reply:
ok